repository/src/main/java/org/alfresco/repo/content/directurl/ContentStoreDirectUrlConfig.java

```diff
@@ -0,0 +1,125 @@
+    {
+        this.maxExpiryTimeInSec = maxExpiryTimeInSec;
+    }
+
+    public Long getMaxExpiryTimeInSec()
+    {
+        return maxExpiryTimeInSec;
+    }
+
+    /**
+     * Configuration initialise
+     */
+    public void init()
+    {
+        validate();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validate()
+    {
+        // Disable direct access URLs for the content store if any error found in the content store direct access URL config
+        try
+        {
+            validateDirectAccessUrlConfig();
+        }
+        catch (InvalidDirectAccessUrlConfigException ex)
+        {
+            logger.error("Disabling content store direct access URLs due to configuration error: " + ex.getMessage());
+            setIsEnabled(false);
+        }
+    }
+
+    /* Helper method to validate the content direct access url configuration settings */
+    private void validateDirectAccessUrlConfig() throws InvalidDirectAccessUrlConfigException
+    {
+        if (isEnabled())
+        {
+            if (getMaxExpiryTimeInSec() == null)
+            {
+                logger.warn("Maximum expiry time property is missing: setting to system-wide maximum.");
+                setMaxExpiryTimeInSec(getSysWideMaxExpiryTimeInSec());
+            }
+            else if (getMaxExpiryTimeInSec() > getSysWideMaxExpiryTimeInSec())
+            {
+                throw new InvalidDirectAccessUrlConfigException("Content store direct access URL maximum expiry time exceeds system-wide maximum expiry time.");
+            }
+
+            if (getDefaultExpiryTimeInSec() == null)
+            {
+                logger.warn("Default expiry time property is missing: setting to system-wide default.");
+                setDefaultExpiryTimeInSec(getSysWideDefaultExpiryTimeInSec());
+            }
+            else if (getDefaultExpiryTimeInSec() > getMaxExpiryTimeInSec())
+            {
+                logger.warn("Default expiry time property exceeds maximum expiry time for content store: setting to system-wide default.");
+                setDefaultExpiryTimeInSec(getSysWideDefaultExpiryTimeInSec());
+            }
+            else if (getDefaultExpiryTimeInSec() > getSysWideDefaultExpiryTimeInSec())
+            {
+                logger.warn("Default expiry time property exceeds system-wide default expiry time: setting to system-wide default.");
+                setDefaultExpiryTimeInSec(getSysWideDefaultExpiryTimeInSec());
+            }
+
+            if (getDefaultExpiryTimeInSec() < 1)
+            {
+                throw new InvalidDirectAccessUrlConfigException("Content store direct access URL default expiry time is invalid.");
+            }
+
+            if (getDefaultExpiryTimeInSec() > getSysWideMaxExpiryTimeInSec())
+            {
+                throw new InvalidDirectAccessUrlConfigException("Content store direct access URL default expiry time exceeds system-wide maximum expiry time.");
```

Comment:
Is it worth providing both values in the message. Would save looking them up.
Only adding a review comment for this one, but there are others as you know.

Reply:
done